pkg/namespace/namespace.go

```diff
@@ -0,0 +1,43 @@
+package namespace
+
+import (
+	"fmt"
+	"strconv"
+
+	corev1Listers "k8s.io/client-go/listers/core/v1"
+)
+
+type NamespaceInfo struct {
+	lister              corev1Listers.NamespaceLister
+	rateLimitAnnotation string
+}
+
+func NewNamespaceInfo(lister corev1Listers.NamespaceLister, rateLimitAnnotation string) *NamespaceInfo {
+	return &NamespaceInfo{
+		lister:              lister,
+		rateLimitAnnotation: rateLimitAnnotation,
+	}
+}
+
+// Retrieves the namespace and extracts the rate-limit annotation value if exists, nil otherwise.
+func (n *NamespaceInfo) RateLimit(namespace string) (*int, error) {
+	if namespace == "" {
+		return nil, fmt.Errorf("invalid namespace; %q", namespace)
+	}
+
+	ns, err := n.lister.Get(namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	if val, ok := ns.Annotations[n.rateLimitAnnotation]; ok {
```

Comment:
This needs to surface data that aligns with the rate limit of argo triggers. The argo rate limit spec is:

```
unit: enum Second, Minute, Hour
requestsPerUnit: int
```

So we probably need a means of normalizing requests per unit to a standard unit or a comparable type that can be used.

20 req / hour and 10 req / second are hard to compare but we can present a standard we want to enforce. "defaults will be in req / sec and limits in other units will be normalized to their per second rate."

Reply:
Thanks. I changed the interface to be
```
RateLimit(namespace string) (*sensor.RateLimit, error)
```

Returns `nil` if the requestsPerUnit is unset.
Returns `Second` for unit if unset.
Otherwise, returns user config.

For normalizing the requests per unit, I think that's better done in the code that does the comparison between default, Namespace, and Trigger (?) value. So I chose to return the RateLimit struct here.
As an aside, I think normalizing to Hour is best. No loss in precision, and int32 max val per `Hour` is enough for 596k/second.